### PR TITLE
Feat isNumber isString 

### PR DESCRIPTION
**Pull request by acamaragl**
### Description:
isNumber
isString

### Breaking changes:
- [ ]

### New features:
- [X]

### Fixes:
- [ ]

### PR status:
- [X] Version bumped
- [X] Change-log updated
- [X] Tests added or updated
- [x] PR keeps 100% test coverage
- [X] Type definition updated
- [X] Readme updated
- [X] Linter is applied
- [X] Const arrow functions are used instead of pure function definitions where applicable
- [X] Functions are listed in alphabetic order in index.js, index.d.ts and readme

### CODE REVIEW

**Review by jsertx — APPROVED** [review]

File: src/isNumber.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'use strict'
+
+const isNumber = (val) => typeof val === 'number'
```
Comment (yaaminu) [inline src/isNumber.js:3]: type of `NaN` is number. Do we want to consider that a valid number for our use cases?

File: src/isString.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+'use strict'
+
+const isString = (val) => typeof val === 'string'
```
Comment (yaaminu) [inline src/isString.js:3]: would fail of strings constructed with `new String(...)`. Is that something we should care about?

**Review by yaaminu — CHANGES_REQUESTED** [review]

File: src/isNumber.js (hunk 1 above)
Comment (jsertx) [reply src/isNumber.js:3]: We are replicating lodash implementation. They return `true` when `NaN` is given.

File: src/isString.js (hunk 2 above)
Comment (jsertx) [reply src/isString.js:3]: Good catch, lets do:

```js
const isString = (val) => typeof val === 'string' || val instanceOf String
```

File: src/isNumber.js (hunk 1 above)
Comment (yaaminu) [reply src/isNumber.js:3]: hmm, if we're replicating lodash then I suggest we copy the code over here. At least the test suit so we know for sure we won't break anything.

Comment (acamaragl) [reply src/isNumber.js:3]: Yep, the idea was to use this instead lodash... but maybe I should delete this method and use directly Number.isFinite in the merchantMap as you said NaN shouldn't be a valid value...

File: src/isFinite.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+'use strict'
+
+const isFinite = (val) => (typeof val === 'number' && Number.isFinite(val)) ||
```
Comment (vigan-abd) [inline src/isFinite.js]: I think this would be simplified like this:
```
const isFinite = (val) => isNumber(val) && Number.isFinite(val)
```

**Review by vigan-abd — CHANGES_REQUESTED** [review]

File: src/isFinite.js (hunk 3 above)
Comment (acamaragl) [reply src/isFinite.js]: hmm , in that case 
`isFinite(new Number(55)) = false ` 
because Number.isFinite(new Number(55)) is false

And it would be true

Comment (bariscicek) [reply src/isFinite.js]: fyi, this is from lodash:
```
   function isFinite(value) {
      return typeof value == 'number' && nativeIsFinite(value);
    }
```
so it is indeed simpler. Tbh, I really don't get the point of reinventing the wheel there, especially there is a pretty solid shinny wheel out there.

Comment (acamaragl) [reply src/isFinite.js]: Hi @bariscicek,
I totally agree with you with 'don't reinvent the wheel' because it is a 'solid shinny wheel', but here I'm having some doubts:
1.- If we don't want to "reinvent the wheel", why don't we fork the current version of lodash under bitfinexcom and use that?
2.- In the isString() and isNumber() methods, some teammates noted that 
`isString( new String('hello') ) ` returned false
`isNumber( new Number (55) ) ` returned false
IMO, we could avoid these cases, although Lodash managed these cases as true... 
but then lodash return false in this:
`_.isFinite(new Number(555))`  ---> false 
In IMO, it is an **error**, and I wanted to fix it in our lib, then what's the point here?
- keep or fix that 'error'? 
- if we don't consider that an error, why should we manage these cases in the isString and isNumber methods?
cc @vigan-abd

Comment (jarco-develop) [reply src/isFinite.js]: If I may. I think is a little overkill to support that kind of cases. We opted for not doing that on the kyc backend https://github.com/bitfinexcom/bfx-core-kyc-js/pull/1017 We used  a simple typeof for string, number and boolean.

Comment (acamaragl) [reply src/isFinite.js]: To close this topic, 
I changed the implementation to Vigan's suggestion.
`(val) => isNumber(val) && Number.isFinite(val)`

In this case, we are aware that 
`isFinite( new Number(24) ) => FALSE` 
we are not very consistent in accepting isNumber(new Number(4)) true and isFinite of that as False... but in any case Lodash did the same...

Comment (vigan-abd) [reply src/isFinite.js]: tbh idk who uses new Number 😅, but indeed seems wrong isFinite being true check against number object, IMHO who uses new Number should use .valueOf to check against number being finite (isFinite(new Number(123).valueOf()))

**Review by vigan-abd — CHANGES_REQUESTED** [review]
@yaaminu @acamaragl @bariscicek @jsertx @jarco-develop I would suggest to remove instanceOf (String, Number...) checks and just use typeof as it seems to be weird behavior supported by js (Number(33) -> type number, new Number(33) -> type object).

Additionally, if we do this we get again type number
console.log(typeof (new Number(33) + new Number(2)));

tldr allowing new Number on code seems wrong to me

**Comment by acamaragl** [comment]
Thank you, Vigan! Done!

**Review by vigan-abd — APPROVED** [review]
